### ISSUE TITLE
1.4.10 Reflow needs a preamble

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#3121]
In discussion with @maryjom about WCAG2ICT work, we noted that Reflow lacks both a preamble giving it context, as well as clearer guidance on how to apply at different form factors.

Setting aside the latter concern for now, I believe that a simple preamble could be added that which would provide guidance for non-web application generally, and potentially for some web applications that are locked down.

The approach is to add a simple preamble, "Where the size of the viewport or the size of the content can be altered...", to the existing text, so that the SC reads:

> Where the size of the viewport or the size of the content can be altered, content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions..."

**Comment by detlevhfischer** [comment]
@mbgower 

> Where the size of the viewport or the size of the content can be altered..."

1. For mobile apps, would you interpret this as being true whenever the orientation of the device (and thereby the viewport width) can be changed?
2. For mobile apps, even where orientation is locked, the platform may (or may not) afford split screen views and thus different viewport dimensions for content. Would this mean that the viewport sitze can be altered in principle even if the actual possibility may depend on the OS of the device where the app is running?

**Comment by detlevhfischer** [comment]
> For question 1, turning a device sideways does not alter the size of the viewport, right?

I guess it may, or may not. With some apps, it appears to be the case that content adapts to a wider viewport and reflows - take the default iOS Mail ap as an example. In other apps, content appears simply enlarged but breaks at the same place.

**Comment by mbgower** [comment]
> I guess it may, or may not. With some apps, it appears to be the case that content adapts to a wider viewport 

Sorry, what I meant is that the size of the window/screen -- does not change by changing the viewport. It is still width x height. it's just that those numbers are inverted. So the overall _size_ of the viewport (what you can see) is not changed.

**Comment by mbgower** [comment]
Confirmed that this issue is included in the [Reflow scratchpad](https://docs.google.com/document/d/1NhXS0Gfdsgl67pIhK0BYmayTq_v45CJMP0Ve36Uui3Q/edit?usp=sharing)

@detlevhfischer if you're interested in being part of that discussion, or joining the WCAG 2.x TF that's dealing with these kinds of things (and meets Fridays at 3PM GMT

**Comment by detlevhfischer** [comment]
@mbgower A the moment I just can't find the time...

**Comment by bruce-usab** [comment]
First impressions, I feel like:
> Where the size of the viewport or the size of the content can be altered...

Is too big a loophole for web content.  I am open to being convinced otherwise.

It does seem perfectly reasonable to me for non-web documents and non-web software (and especially platform-specific software).

**Comment by maryjom** [comment]
I think you have to look at various use cases for viewing web content:
- Where web content is designed for use on a phone (e.g. a certain presentation for phones using responsive design) - Some phones may not support the viewport CSS pixel width or height as specified by WCAG in CSS pixels. Should one test to the nearest supported height or width as the case might be? The SC and guidance don't cover that situation.  Is its scrollable viewport allowed to be larger than the specified CSS pixels? Is it allowed to be smaller?
- What happens when a user agent doesn't support that particular width or height for scrolling? E.g. when a user agent on a tablet provides multiple web pages to be viewed at a time side-by-side.  The scrollable viewport may not be able to be viewed at the scrolling width or height specified by WCAG. Should there be a tolerance, or a more generic application of the SC, or not be a hard requirement at that point? 
- What about a web TV? Is a user able to adjust the viewport?
- Consider a desktop-based web browser. I don't think they always support resizing the window to the scrolling width or height specified. 

Perhaps some notes could be written to the effect that where a user agent doesn't support the scrolling width or height specified, that the web content will scroll within the size the user agent supports.

**Comment by gundulaniemann** [comment]
> First impressions, I feel like:
> 
> > Where the size of the viewport or the size of the content can be altered...
> 
> Is too big a loophole for web content. I am open to being convinced otherwise.
> 
> It does seem perfectly reasonable to me for non-web documents and non-web software (and especially platform-specific software).

I must admit I do not see the loophole.
Web content runs in a browser, and the browser window sine can always be chnaged, and the browser provides functionality to change text size.
